src/api.js

```diff
@@ -662,6 +664,7 @@ api.declare({
     var task = await this.Task.create({
       taskId:             taskId,
       provisionerId:      taskDef.provisionerId,
+      clientId:           taskDef.clientId || clientId_default, 
```

Comment:
@fiennyangeln we need to check that `taskDef.clientId === clientId_default` first, otherwise a caller could "lie" about their clientId.

@petemoore this seems a bit silly.  Why not just fill the field in with `req.clientId` unconditionally, instead of allowing the user to specify it in taskDef?